satpy/tests/writer_tests/test_mitiff.py

```diff
@@ -241,6 +275,25 @@ def test_save_dataset_with_calibration(self):
         w = MITIFFWriter(filename=dataset.attrs['metadata_requirements']['file_pattern'], base_dir=self.base_dir)
         w.save_dataset(dataset)
 
+    def test_save_dataset_with_bad_value(self):
+        """Test basic writer operation."""
+        import os
+        import numpy as np
+        from libtiff import TIFF
+        from satpy.writers.mitiff import MITIFFWriter
+
+        expected = np.array([[0, 4, 1, 37, 73],
+                             [110, 146, 183, 219, 255]])
+
+        dataset = self._get_test_dataset_with_bad_values()
+        w = MITIFFWriter(base_dir=self.base_dir)
+        w.save_dataset(dataset)
+        filename = "{:s}_{:%Y%m%d_%H%M%S}.mitiff".format(dataset.attrs['name'],
+                                                         dataset.attrs['start_time'])
+        tif = TIFF.open(os.path.join(self.base_dir, filename))
+        for image in tif.iter_images():
+            np.testing.assert_allclose(image, expected, atol=1.e-6, rtol=0)
+
```

Comment:
Could you delete the file when the test is finished ? Or do you do that later ?

Reply:
I think cleanup does that. Will check

Reply:
tearDown removes the temp directory.

Reply:
Perfect. Merging.